sql/src/main/java/org/apache/druid/sql/avatica/DruidMeta.java

```diff
@@ -446,15 +446,16 @@ public MetaResultSet getColumns(
     }
 
     if (schemaPattern.s != null) {
-      whereBuilder.add("COLUMNS.TABLE_SCHEMA LIKE " + Calcites.escapeStringLiteral(schemaPattern.s));
```

Comment:
This looks like the right fix. Could you please make it to all of `getSchemas`, `getTables`, and `getColumns`? They have the same problem. It might be nice to create a private static helper function in this file to avoid all the boilerplate, too.

Reply:
Thanks for taking a look, @gianm . Pushed a commit to add the `ESCAPE` clause in the three methods along with tests.